File: web/src/main/java/top/bettercode/summer/web/serializer/KilogramDeserializer.java
package top.bettercode.summer.web.serializer;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import java.io.IOException;
import top.bettercode.summer.web.support.KilogramUtil;

/**
 * @author Peter Wu
 */
public class KilogramDeserializer extends JsonDeserializer<Long> {

  @Override
  public Long deserialize(JsonParser p, DeserializationContext ctxt)
      throws IOException {
    return KilogramUtil.toGram(p.getValueAsString());
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/serializer/KilogramDeserializer.kt
package top.bettercode.summer.web.serializer

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import top.bettercode.summer.web.support.KilogramUtil.toGram

/**
 * @author Peter Wu
 */
class KilogramDeserializer : JsonDeserializer<Long>() {
    override fun deserialize(p: JsonParser, ctxt: DeserializationContext): Long {
        return toGram(p.valueAsString)
    }
}
